operations/helm/charts/mimir-distributed/templates/store-gateway/store-gateway-statefulset.yaml

```diff
@@ -1,27 +1,36 @@
+{{- $args := dict "ctx" $ "component" "store-gateway" "memberlist" true -}}
+{{- $zonesMap := include "mimir.zoneAwareReplicationMap" $args | fromYaml -}}
+{{- range $zoneName, $rolloutZone := $zonesMap }}
+{{- with $ -}}
+{{- $_ := set $args "rolloutZoneName" $zoneName -}}
 apiVersion: apps/v1
 kind: StatefulSet
 metadata:
-  name: {{ include "mimir.resourceName" (dict "ctx" . "component" "store-gateway") }}
+  name: {{ include "mimir.resourceName" $args }}
   labels:
-    {{- include "mimir.labels" (dict "ctx" . "component" "store-gateway" "memberlist" true) | nindent 4 }}
+    {{- include "mimir.labels" $args | nindent 4 }}
   annotations:
-    {{- toYaml .Values.store_gateway.annotations | nindent 4 }}
+    {{- include "mimir.componentAnnotations" $args | nindent 4 }}
   namespace: {{ .Release.Namespace | quote }}
 spec:
-  replicas: {{ .Values.store_gateway.replicas }}
+  replicas: {{ $rolloutZone.replicas }}
   selector:
     matchLabels:
-      {{- include "mimir.selectorLabels" (dict "ctx" . "component" "store-gateway" "memberlist" true) | nindent 6 }}
+      {{- include "mimir.selectorLabels" $args | nindent 6 }}
   updateStrategy:
+    {{- if $zoneName }}
+    type: OnDelete
+    {{- else }}
     {{- toYaml .Values.store_gateway.strategy | nindent 4 }}
```

Comment:
Can you leave a comment on the values.yaml that `store_gateway.strategy` is not taken into account if using zone aware replication? Same goes for ingesters and alertmanagers. In their case I think it's statefulStrategy instead of strategy.